beater/v2_handler_test.go

```diff
@@ -19,17 +19,37 @@ package beater
 
 import (
 	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
 	"net/http"
 	"net/http/httptest"
 	"testing"
 
+	"github.com/elastic/apm-server/publish"
+	"github.com/elastic/apm-server/tests"
+
 	"github.com/pkg/errors"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	"github.com/elastic/apm-server/decoder"
+	"github.com/elastic/apm-server/tests/loader"
 	"github.com/elastic/apm-server/transform"
 )
 
+func approveResultBody(t *testing.T, name string, body []byte) {
+	var resultmap map[string]interface{}
+	err := json.Unmarshal(body, &resultmap)
+	assert.NoError(t, err, "err", string(body))
+
+	resultName := fmt.Sprintf("approved-stream-result/%s", name)
+	verifyErr := tests.ApproveJson(resultmap, resultName, nil)
+	if verifyErr != nil {
+		assert.Fail(t, fmt.Sprintf("Test %s failed with error: %s", name, verifyErr.Error()))
+	}
+}
```

Comment:
Would it make sense to move `approveResultBody` to the `test` package? I think I saw this copy&pasted a few times in former PR already.

Reply:
refactored a bit. WDYT?